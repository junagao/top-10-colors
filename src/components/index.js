export { default as ColorList } from 'components/ColorList'
export { default as ColorItem } from 'components/ColorItem'
export { default as Header } from 'components/Header'
export { default as Rating } from 'components/Rating'
export { default as Star } from 'components/Star'
