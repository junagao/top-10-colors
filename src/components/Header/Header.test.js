import React from 'react'
import { shallow } from 'enzyme'
import { Header, Button } from 'components'

describe('Header', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <Header
        isRandomRating
        startRandomRating={jest.fn()}
        stopRandomRating={jest.fn()}
        onResetRatings={jest.fn()}
      />,
    )
  })

  it('should render correctly given the required props', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render one HeaderContainer element', () => {
    expect(wrapper.find('HeaderContainer').length).toEqual(1)
  })

  it('should render one HeaderTitle element with the title', () => {
    expect(wrapper.find('HeaderTitle').length).toEqual(1)
    expect(wrapper.find('HeaderTitle').children().text()).toEqual(
      'Top 10 colors',
    )
  })

  it('should render one div element', () => {
    expect(wrapper.find('div').length).toEqual(1)
  })

  it('should render two instance of Button component', () => {
    expect(wrapper.find(Button).length).toEqual(2)
  })

  it('should render one instance of Button component with prop `text` equal to stop if isRandomRating true', () => {
    expect(wrapper.find(Button).at(0).prop('text')).toEqual('stop rating')
  })

  it('should render one instance of Button component with prop `text` equal to `random rate` if isRandomRating false', () => {
    wrapper.setProps({ isRandomRating: false })
    expect(wrapper.find(Button).at(0).prop('text')).toEqual('random rate')
  })

  it('should render one instance of Button component with prop `text` equal to `reset ratings`', () => {
    expect(wrapper.find(Button).at(1).prop('text')).toEqual('reset ratings')
  })
})
