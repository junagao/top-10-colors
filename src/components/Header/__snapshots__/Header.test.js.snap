// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly given the required props 1`] = `
<HeaderContainer>
  <HeaderTitle>
    Top 10 favorite colors
  </HeaderTitle>
  <div>
    <Button
      onClick={[MockFunction]}
      reset={false}
      text="stop"
    />
    <Button
      onClick={[MockFunction]}
      reset={true}
      text="reset ratings"
    />
  </div>
</HeaderContainer>
`;
